src/test/java/gift/controller/WishControllerTest.java

```diff
@@ -103,14 +108,18 @@ void createWishTest() {
     void deleteWishTest() {
 
         // given
-        Category category = new Category("name", "color", "imageUrl", "description");
-        Category savedCategory = categoryRepository.save(category);
-
-        Product product = new Product("name", 1000, "imageUrl", savedCategory);
-        productRepository.save(product);
-
-        Wish wish = new Wish(memberService.getMemberFromToken(token), product);
-        wishRepository.save(wish);
+        TransactionTemplate transactionTemplate = new TransactionTemplate(transactionManager);
+        transactionTemplate.execute(status -> {
+            Category category = new Category("name", "color", "imageUrl", "description");
+            Category savedCategory = categoryRepository.save(category);
+
+            Product product = new Product("name", 1000, "imageUrl", savedCategory);
+            productRepository.save(product);
+            Wish newWish = new Wish(memberService.getMemberFromToken(token), product);
+            wishRepository.save(newWish);
+
+            return null;
+        });
```

Comment:
> 처음에는 트랜잭션 전파 기능을 사용하여 getMemberFromToken의 트랜잭션과 테스트의 트랜잭션을 묶어버리면 영속성 컨텍스트가 유지될 것이라고 생각했습니다. 그래서 테스트 메서드에 선언적 트랜잭션 처리를 적용해보았습니다. 예상대로 초기화 오류는 발생하지 않았지만, 테스트는 실패했습니다.
> 
> 그 이유를 생각해보니, 테스트 메서드에 @transactional 처리를 하면 트랜잭션 내부의 save가 쓰기 지연으로 인해 테스트 메서드가 종료된 후에 실행되므로 restTemplate으로 실행 중에는 DB에 값이 없다고 나오는 것으로 추측했습니다.
> 
> 이를 해결하기 위해 여러 레퍼런스를 찾아본 결과, TransactionTemplate를 사용하면 트랜잭션 범위를 설정할 수 있다는 것을 알게 되었습니다. 그래서 트랜잭션 범위를 시작부터 wish를 save 하는 것까지 묶었더니 예상대로 오류를 해결할 수 있었습니다.
> 
> 저는 트랜잭션 범위를 조정하여 오류를 해결했지만, 다른 방법도 있을지 궁금합니다.


학습 잘 해주셨네요 👍 트랜잭션 매니저를 사용하지 않고 해볼 수 있는 다른 방법은 
`wishRepository.save(newWish);` 대신  `wishService.createWish(wishRequest);`을 사용하는 겁니다. 
```suggestion
        Category category = new Category("name", "color", "imageUrl", "description");
        Category savedCategory = categoryRepository.save(category);

        Product product = new Product("name", 1000, "imageUrl", savedCategory);
        productRepository.save(product);

        Member member = memberService.getMemberFromToken(token);
        WishRequest wishRequest = new WishRequest(member.getId(), product.getId());
        wishService.createWish(wishRequest);
```

Reply:
감사합니다!! servcie 계층을 이용하면 쉽게 해결할 수 있는 문제였군요!
 
이렇게 작성하면 createWish 메서드 안의 DtoToEntity 메서드에서 memberRepository를 통해 Member를 가져올 때 createWish의 트랜잭션이 전파되어 repository의 트랜잭션과 합쳐지기 때문에 영속성 컨텍스트의 생명 주기가 늘어나 지연 로딩(lazy loading) 초기화 오류가 발생하지 않는 것으로 이해했는데, 제가 이해한 게 맞을까요??